File: Python/백준/Silver/10828. 스택/스택.py
import sys

N = int(sys.stdin.readline()) # 명령의 수
stack = []
for _ in range(N):
    command = sys.stdin.readline().strip()

    if command[:4] == 'push':
        _, x = command.split()
        stack.append(int(x))
    elif command == 'pop':
        if stack:
            print(stack.pop())
        else:
            print(-1)
    elif command == 'size':
        print(len(stack))
    elif command == 'empty':
        if stack:
            print(0)
        else:
            print(1)
    elif command == 'top':
        if stack:
            print(stack[-1])
        else:
            print(-1)

File: node.js/백준/Silver/10828. 스택/스택.js
const fs = require("fs");
const input = fs.readFileSync("/dev/stdin").toString().trim().split("\n");

const stack = [];
const result = [];

const N = parseInt(input[0]);

for (let i = 1; i <= N; i++) {
    const [command, value] = input[i].split(" ");
    
    switch (command) {
        case "push":
            stack.push(parseInt(value));
            break;
        case "pop":
            result.push(stack.length ? stack.pop() : -1);
            break;
        case "size":
            result.push(stack.length);
            break;
        case "empty":
            result.push(stack.length ? 0 : 1);
            break;
        case "top":
            result.push(stack.length ? stack[stack.length-1] : -1);
            break;
    }
}

console.log(result.join("\n"));